File: brasilburger-java/src/main/java/sn/ism/brasilburger/model/Quartier.java
package sn.ism.brasilburger.model;

public class Quartier {
    private int id;
    private String libelle;
    private Zone zone; 

    public Quartier() {
    }

    public Quartier(int id, String libelle, Zone zone) {
        this.id = id;
        this.libelle = libelle;
        this.zone = zone;
    }

    public Quartier(String libelle, Zone zone) {
        this(0, libelle, zone);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public Zone getZone() {
        return zone;
    }

    public void setZone(Zone zone) {
        this.zone = zone;
    }

    @Override
    public String toString() {
        return "Quartier{" +
                "id=" + id +
                ", libelle='" + libelle + '\'' +
                ", zone=" + (zone != null ? zone.getLibelle() : "null") +
                '}';
    }
}

File: src/Entity/Quartier.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'quartier')]
class Quartier
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_quartier', type: 'integer')]
    private ?int $idQuartier = null;

    #[ORM\Column(type: 'string', length: 120)]
    private string $libelle;

    #[ORM\ManyToOne(targetEntity: Zone::class, inversedBy: 'quartiers')]
    #[ORM\JoinColumn(name: 'id_zone', referencedColumnName: 'id_zone', nullable: false)]
    private Zone $zone;

    public function getIdQuartier(): ?int { return $this->idQuartier; }
    public function getLibelle(): string { return $this->libelle; }
    public function setLibelle(string $libelle): self { $this->libelle = $libelle; return $this; }
    public function getZone(): Zone { return $this->zone; }
    public function setZone(Zone $zone): self { $this->zone = $zone; return $this; }
}
